### PR TITLE
Feature/artists

### PR DESCRIPTION
**Pull request by larkinds**
Artists treeview & commands are implemented & some comments cleanup for playlists & albums

### CODE REVIEW

File: src/artists.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      this.contextValue = 'artistTrack';
+    }
+
+  }
+
+  async getTracks(id: string, spotify: SpotifyWebApi) {
+    let tracks = await spotify.getArtistTopTracks(id, 'US');
+    return tracks.body.tracks.map((track: any) => new ArtistsAndTracks(track));
+  }
+}
```
Comment (jkmartindale) [inline src/artists.ts]:
```suggestion
}

```

File: src/artists.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+export class ArtistsAndTracks extends vscode.TreeItem {
+  name: string;
+  uri: string;
+  children: Promise<ArtistsAndTracks[]> | undefined;
+
+  constructor (public readonly data: any, spotify?: SpotifyWebApi, parent?: ArtistsAndTracks) {
+    super(data);
```
Comment (jkmartindale) [inline src/artists.ts]: I'm assuming this is the proper label given the code below
```suggestion
    super(data.name);
```

File: src/artists.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    if (spotify) {
+      this.name = data.name;
+      this.uri = '';
+      this.children = this.getTracks(data.id, spotify);
+
+      this.description = data.name;
```
Comment (jkmartindale) [inline src/artists.ts]: Do we need to set a description if the label is set properly?

File: src/artists.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      this.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+      this.contextValue = 'artist';
+    } else {
+      this.name = data.name;
+      this.uri = data.uri;
+      this.children = undefined;
+
+      this.description = `${data.name}`;
```
Comment (jkmartindale) [inline src/artists.ts]: Same as previous comment

**Review by jkmartindale — CHANGES_REQUESTED** [review]
Give me my empty line at the end of a file, I wanna see real item labels, merge this in after #31, branch off `staging` next time, blah blah blah

File: src/artists.ts (hunk 3 above)
Comment (larkinds) [reply src/artists.ts]: 💯

**Pull request merged**